packages/sdk-js/src/GrowthBook.ts

```diff
@@ -987,7 +1082,40 @@ export class GrowthBook<
     }
 
     // 13. Build the result object
-    const result = this._getResult(experiment, assigned, true, featureId, n);
+    const result = this._getResult(
+      experiment,
+      assigned,
+      true,
+      featureId,
+      n,
+      foundStickyBucket
+    );
+
+    // 13.5. Persist sticky bucket
+    if (this.context.enableStickyBucketing && experiment.stickyBucketing) {
+      const {
+        changed,
+        key: attrKey,
+        doc,
+      } = this._generateStickyBucketAssignmentDoc(
+        hashAttribute,
+        toString(hashValue),
+        {
+          [this._getStickyBucketExperimentKey(
+            experiment.key,
+            experiment.bucketVersion
+          )]: result.variationId,
+        }
+      );
+      if (changed) {
+        // update local docs
+        this._ctx.stickyBucketAssignmentDocs =
+          this._ctx.stickyBucketAssignmentDocs ?? {};
+        this._ctx.stickyBucketAssignmentDocs[attrKey] = doc;
+        // save doc
+        this._ctx.stickyBucketService?.saveAssignments(doc);
```

Comment:
todo: determine if the `?.` operator causes bundle size inefficiencies during transpilation

Reply:
The `?.` operator technically checks if the value is either undefined or null, so the transpilation is pretty inefficient.  It will basically do:

```ts
(this._ctx.stickyBucketService === null || this._ctx.stickyBucketService === undefined) 
  ? undefined 
  : this._ctx.stickyBucketService.saveAssignments(doc)
```

Shorter to replace with:

```ts
this._ctx.stickyBucketService && this._ctx.stickyBucketService.saveAssignments(doc)
```

Not too bad when there's a single optional chain, but gets really verbose if you have multiple ones in a statement.